File: src/main/java/no/nav/bidrag/beregn/nettobarnetilsyn/periode/NettoBarnetilsynPeriode.java
package no.nav.bidrag.beregn.nettobarnetilsyn.periode;

import java.util.List;
import no.nav.bidrag.beregn.felles.bo.Avvik;
import no.nav.bidrag.beregn.nettobarnetilsyn.beregning.NettoBarnetilsynBeregning;
import no.nav.bidrag.beregn.nettobarnetilsyn.bo.BeregnNettoBarnetilsynGrunnlag;
import no.nav.bidrag.beregn.nettobarnetilsyn.bo.BeregnetNettoBarnetilsynResultat;

public interface NettoBarnetilsynPeriode {
  BeregnetNettoBarnetilsynResultat beregnPerioder(BeregnNettoBarnetilsynGrunnlag beregnNettoBarnetilsynGrunnlag);

  List<Avvik> validerInput(BeregnNettoBarnetilsynGrunnlag beregnNettoBarnetilsynGrunnlag);

  static NettoBarnetilsynPeriode getInstance() {
    return new NettoBarnetilsynPeriodeImpl(NettoBarnetilsynBeregning.getInstance());
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/nettobarnetilsyn/periode/NettoBarnetilsynPeriode.kt
package no.nav.bidrag.beregn.nettobarnetilsyn.periode

import no.nav.bidrag.beregn.felles.bo.Avvik
import no.nav.bidrag.beregn.nettobarnetilsyn.beregning.NettoBarnetilsynBeregning
import no.nav.bidrag.beregn.nettobarnetilsyn.bo.BeregnNettoBarnetilsynGrunnlag
import no.nav.bidrag.beregn.nettobarnetilsyn.bo.BeregnetNettoBarnetilsynResultat

interface NettoBarnetilsynPeriode {
    fun beregnPerioder(grunnlag: BeregnNettoBarnetilsynGrunnlag): BeregnetNettoBarnetilsynResultat

    fun validerInput(grunnlag: BeregnNettoBarnetilsynGrunnlag): List<Avvik>

    companion object {
        fun getInstance(): NettoBarnetilsynPeriode = NettoBarnetilsynPeriodeImpl(NettoBarnetilsynBeregning.getInstance())
    }
}
